src/mainboard/pcengines/apu2/OemCustomize.c

```diff
@@ -132,31 +132,31 @@ static const PCIe_PORT_DESCRIPTOR PortList[] = {
 };
 
 static const PCIe_PORT_DESCRIPTOR PortListAspmReverse[] = {
-	/* Initialize Port descriptor (PCIe port, Lanes 3, PCI Device Number 4, ...) */
+	/* Initialize Port descriptor (PCIe port, Lanes 3, PCI Device Number 2.5, ...) */
 	{
 		0,
 		PCIE_ENGINE_DATA_INITIALIZER(PciePortEngine, 3, 3),
-		PCIE_PORT_DATA_INITIALIZER_V2(PortEnabled, ChannelTypeExt6db, 2, 4,
+		PCIE_PORT_DATA_INITIALIZER_V2(PortEnabled, ChannelTypeExt6db, 2, 5,
```

Comment:
I would like to avoid this change on apu2, because it is mPCIe1 slot, which should have 2.4 and mPCIe2 slot should be 2.5 (lanes 4-7). mPCIe2 slot has troubles with PCIe module detection. If something is not detected, then mPCIe1 interface address will also change in such a situation

Reply:
Soo, again another const array has to be added, this time for apu2? It's getting crowded there ^^

Reply:
Well you may play a little bit with compiler and try to remove `const` and then apply const when passing to the AGESA parameters structure. Maybe @krystian-hebel would give you some hints how to cheat compiler. Then you coudl simpyl modify only single fields in the function

Reply:
This is romstage I presume? In that case, fooling a compiler may not be enough if the data is still accessed directly from ROM. It may be better to do `#ifdef BOARD_PCENGINES_APUx`, if not directly in this struct (for better readability) maybe `#define` a constant before this.

Reply:
@miczyg1 If mPCIe1 should be 2.4 shouldn't default PCIe order be changed as well? Without reversing mPCIe1 is 2.5 and mPCIe2 is 2.1.

Reply:
NO, we cannot touch the default order, if we could, we wouldn't bother with runtime config option